### ISSUE TITLE
Brazil FITCount app not uploading

### ISSUE THREAD

**Issue by JimChiazzese1** [NERC-CEH/fit-count-app#139]
Brazil users are reporting that they are unable to upload the surveys they've been taking in the last couple of days.
They can take the survey, but when they hit the icon / button “carrega”, it appears it's trying to do something, but nothing happens. I've asked them to confirm if they're trying to upload over WiFi or a strong connection.
Please let me know what we can do to trouble shoot.

**Comment by JimChiazzese1** [comment]
FYI - The Brazil team reporting the issue are all using iphone/iOS and have tried uploading counts both on wifi and 4g. I have just tried a FITCount on a google/android device, using the Brazil version of the app (in English) and placing my location at a random spot in Brazil and had no issue. The count uploaded and appears fine on FITCount.ceh.ac.uk. Suspect it may be an apple issue...

**Comment by kazlauskis** [comment]
Can you please check which app version they are using? Also, ideally I would need a timestamp of the unsuccessful upload so that I can look into the server logs.

**Comment by JimChiazzese1** [comment]
Thanks Karolis. Will ask them to confirm. Regarding timestamp - Is there anyway to see that from the pending survey? The only detail I can see on the survey is the date. Assume you mean timestamp of exactly when they tried to upload (sorry, reread your comment, that makes more sense). Will request Brazil try and upload again and record exactly what time and send that through. Thanks

**Comment by JimChiazzese1** [comment]
So far just details of one user account - saraiva@usp.br, user Antonio Saraiva, however he also states he's on version 1.1.3 (144), whereas I'm on v1.2.0(149) - I assume the android version automatically updates by default, whereas the iOS version may not?

**Comment by kitenetter** [comment]
I can confirm that Antonio's account on the FIT Count website is currently missing a first and last name, which for UK users seems to result in the app failing to upload. @JimChiazzese1 do you want me to add the first and last name to his account, and then he can try again?

**Comment by JimChiazzese1** [comment]
Yes, please. Thanks Karolis.
Would there be a simple fix to this, eg making those fields mandatory?

**Comment by kitenetter** [comment]
@JimChiazzese1 I've updated the fields for Antonio, and I will be very interested to find out if that now allows the records to be uploaded.

As to whether there is a simple fix, we're awaiting input from Karolis for issue #137 (but I think he's on leave?).

**Comment by JimChiazzese1** [comment]
Apologies @kitenetter and @kazlauskis - I didn't notice who had made which comments, and assumed it was all Karolis. Karolis is on leave but has been answering to comments. I'll request the user tries upload again.

**Comment by JimChiazzese1** [comment]
It's been verified by a UK user that logging in/out (and unchecking the relevant box) will work again once their name is updated on their account. This has been done on our end.
However, if a user goes to FITCount.ceh.ac.uk and updates their profile themselves (presuming missing name) and then logs out and back in - will that have the same effect as us (or rather @kitenetter) updating their details? 
If so, then I can alert the country coordinators to this and they can pass on the workaround to their user. Thanks

**Comment by andrewvanbreda** [comment]
@kazlauskis 
These fields are installed automatically installed by Indicia's Easy Login module, and that module is where my new code is contained.

So they are not part of the new code, however they are part of the standard Indicia easy_login Drupal module that has been installed.

Normally Indicia easy_login identifies the user from the warehouse using the last name in part.

Deleting could cause issues, I can see code on login that will warn the user if the last_name isn't filled in.

I am not sure what to suggest as this is quite unusual situation.

I suppose the only thing I can think of is to move my new deletion code into a new Drupal module which only has that deletion function,
then delete those troublesome fields.

I am trying to think on the fly here.......I think that might work......probably....maybe

**Comment by andrewvanbreda** [comment]
@kazlauskis 
Does your app code fill in the Drupal Indicia Warehouse Id field when an account is created? (field_indicia_user_id)

**Comment by andrewvanbreda** [comment]
@kazlauskis 
Sorry another question. If you don't have separate first name and surname fields, what is set in the Warehouse people table which has those fields separately?
Cheers

**Comment by kazlauskis** [comment]
There is a single `field_full_name` value that is used for the warehouse ID. I agree it would be better to separate any account deletion functionality into its own module.

**Comment by kazlauskis** [comment]
@kitenetter @JimChiazzese1 I have now fixed the website module to ignore the new user name fields and also fixed any user accounts affected by this (created in the last 2 weeks). This means that any newly registered as well as existing users can now upload surveys. Existing users might need to wait up to a day for their login session to renew - tomorrow it should work fine for everybody. Otherwise, these users can also sign out and sign in back again to fix this right away. 

I have moved Andy's part to another ticket.
